npe2/_plugin_manager.py

```diff
@@ -135,14 +135,20 @@ def iter_compatible_readers(self, paths: List[str]) -> Iterator[ReaderContributi
             return
         assert isinstance(path, str)
 
+        # use lower() to make matching case-insensitive
+        path = path.lower()
+
         if os.path.isdir(path):
```

Comment:
on UNIX systems (MacOS, Linux) filesystem is cases sensitive:

`/Users/` will be converted to `/users/` which most probably does not exists.

Reply:
Hmm, i'm on macOS and it still matches correctly. 
But yeah, probably safer to use something like:
```
base, ext = os.path.splitext(path)
path = os.path.join(base + ext.lower())
```

Reply:
both `os.path.isdir` and os.path.splitext` work on `.lower()` even though the real paths on my computer have cases.

Reply:
Can instead just move the `.lower()` to the `fnmatch` here:
https://github.com/psobolewskiPhD/npe2/blob/b6cf2cf8d72ff287051b1e02cd79f9a8e537ec0f/npe2/_plugin_manager.py#L150

Reply:
OK, so this actually does nothing, because `paths` is set to lower in `io_utils.py` `_read` (see below).
So some duplication on my part. Not sure where it's more logical to do this.
`io_utils.py`: `_read` or `read_get_reader`
or here in `_plugin_manager` (deleting the changes in `io_utils`)

Reply:
Ok, i avoid the duplication now. `lower()` is used in `_read` only, I think this makes more sense.

Reply:
OK, that was annoying because _read can take a str or a list.
I put the check back in _plugin_manager, in the correct _iter_compatible_readers
(sorry made a mess of the commits!)